client/src/pages/feedback.tsx

```diff
@@ -116,16 +132,89 @@ const columnHeaders: ColumnDef[] = [
   },
 ];
 
+function formatMentorQuestions(
+  mentorAnswers: Answer[],
+  mentorQuestions: Record<string, QuestionState>
+) {
+  const completeAnswers = mentorAnswers
+    .filter((mentorAnswer) => mentorAnswer.status == Status.COMPLETE)
+    .sort((a, b) =>
+      (mentorQuestions[a._id].question?.question || "") >
+      (mentorQuestions[b._id].question?.question || "")
+        ? 1
+        : (mentorQuestions[a._id].question?.question || "") <
+          (mentorQuestions[b._id].question?.question || "")
+        ? -1
+        : 0
+    );
+
+  const incompleteAnswers = mentorAnswers
+    .filter((mentorAnswer) => mentorAnswer.status == Status.INCOMPLETE)
+    .sort((a, b) =>
+      (mentorQuestions[a._id].question?.question || "") >
+      (mentorQuestions[b._id].question?.question || "")
+        ? 1
+        : (mentorQuestions[a._id].question?.question || "") <
+          (mentorQuestions[b._id].question?.question || "")
+        ? -1
+        : 0
+    );
+
+  return completeAnswers.concat(incompleteAnswers);
+}
+
 function FeedbackItem(props: {
+  openModal: boolean;
+  setOpenModal: (openModal: boolean) => void;
+  mentor: Mentor;
+  accessToken?: string;
   feedback: UserQuestion;
   mentorAnswers?: Answer[];
   mentorQuestions: Record<string, QuestionState>;
   onUpdated: () => void;
+  queueList: string[];
+  setQueueList: (queueList: string[]) => void;
 }): JSX.Element {
-  const { feedback, mentorAnswers, mentorQuestions, onUpdated } = props;
+  const {
+    openModal,
+    setOpenModal,
+    mentor,
+    accessToken,
+    feedback,
+    mentorAnswers,
+    mentorQuestions,
+    onUpdated,
+    queueList,
+    setQueueList,
+  } = props;
+  const [selectedAnswerStatus, setSelectedAnswerStatus] = useState<Status>(); // for disabling/enabling queue button
+  const [selectedAnswerID, setSelectedAnswerID] = useState<string>();
+
+  // function to add/remove from queue
+  async function queueButtonClicked(
+    selectedAnswerID: string,
+    accessToken: string
+  ) {
+    if (queueList?.includes(selectedAnswerID)) {
+      setQueueList(
+        await removeQuestionFromRecordQueue(selectedAnswerID, accessToken)
+      );
+    } else if (selectedAnswerID == undefined || selectedAnswerID == "") {
+      setOpenModal(true);
```

Comment:
This should have a better name, maybe "setCustomQuestionModalOpen", if that's what it's doing, not sure